pkg/restapi/issuer/operation/operations.go

```diff
@@ -464,14 +471,50 @@ func (c *Operation) preAuthorize(w http.ResponseWriter, r *http.Request) {
 		initiateReq.UserPinRequired = false
 	}
 
-	c.buildInitiateOIDC4CIFlowPage(w, profile.ID, initiateReq, c.preAuthorizeHTML)
+	c.buildInitiateOIDC4CIFlowPage(w, profile.ID, initiateReq, c.preAuthorizeHTML, "")
+}
+
+func (c *Operation) openID4CIHandler(w http.ResponseWriter, r *http.Request) {
+	profile, err := c.determineProfile(r.URL.Query())
+	if err != nil {
+		c.writeErrorResponse(w, http.StatusBadRequest, err.Error())
+		return
+	}
+
+	initiateReq := &initiateOIDC4CIRequest{}
+
+	flowType := c.determineFlowType(r.URL.Query())
+
+	if flowType == "pre-authorized" {
+		initiateReq = &initiateOIDC4CIRequest{
+			CredentialTemplateID: profile.CredentialTemplateID,
+			UserPinRequired:      true,
+			ClaimData:            &profile.Claims,
+		}
+
+		if strings.EqualFold(r.URL.Query().Get("require_pin"), "false") {
```

Comment:
Just a minor coding style note: I would suggest removing line 491, which will result in the `UserPinRequired` field defaulting to false. Then, on line 495 check for `"true"` instead. This way, you save one line. Also, if "require_pin" is missing from the URL query for whatever reason, you'll default to false (which is generally what you should default to).

Unless you intended to have `UserPinRequired` default to true if "require_pin" is missing from the URL query, in which case you can disregard this comment :)